src/terminal/terminal.c

```diff
@@ -1590,17 +1590,112 @@ static int __guac_terminal_send_key(guac_terminal* term, int keysym, int pressed
                 if (keysym == 0xFF54 || keysym == 0xFF99) return guac_terminal_send_string(term, "\x1B[B"); /* Down */
             }
 
+            /* Check if any Modifier key is pressed. */
+            if (term->mod_ctrl || term->mod_alt || term->mod_shift) {
+                int keyseq = 0;
+
+                if (term->mod_ctrl)     keyseq = keyseq | 4; /* OR operation with other modifier key combination. */
+                if (term->mod_alt)      keyseq = keyseq | 2; 
+                if (term->mod_shift)    keyseq = keyseq | 1;
+
+                keyseq = keyseq + 1;
+                
+                /* F1  */
+                if (keysym == 0xFFBE || keysym == 0xFF91) {
+                    char data[10];
+                    sprintf(data, "\x1B[1;%dP", keyseq);
+                    return guac_terminal_send_string(term, data); 
```

Comment:
You can avoid this by using `guac_terminal_sendf()`:

https://github.com/apache/guacamole-server/blob/73ac4fcdbe5116a0eb5c5899af1c13ccb7be11d8/src/terminal/terminal/terminal.h#L974-L995

Reply:
Like you are suggesting like this maybe. Like this
```
if(any mmodifier pressed)
//keyseq calculated
guac_terminal_sendf(term, "\x1B[1;"); 
char charValue = keyseq+'0';
guac_terminal_sendf(term, charValue);
//Till here can be removed outside each if like repetitions can be removed
if(keypress == 0xFFBE) //
return guac_terminal_sendf(term, "P");  //Like for F1
```

Reply:
No. From the documentation above, `guac_terminal_sendf()` accepts a printf-style format string and list of arbitrary arguments. Where you have:

```c
char data[10];
sprintf(data, "\x1B[1;%dP", keyseq);
return guac_terminal_send_string(term, data);
```

you can instead:

```c
return guac_terminal_sendf(term, "\x1B[1;%dP", keyseq);
```

Reply:
Done.